docs/_howto/lolex-async-promises.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: page
+title: How to test async functions with fake timers
+---
+
+With fake timers (lolex), testing code that depends on timers is easier, as it sometimes
+becomes possible to skip the waiting part and trigger scheduled callbacks
+synchronously. Consider the following function of a maker module (a module
+that makes things):
+
+```js
+// maker.js
+module.exports.callAfterOneSecond = callback => {
+    setTimeout(callback, 1000);
+};
+```
+
+We can use Mocha with lolex to verify that `callAfterOneSecond` works as expected, but
+skipping that part where the test takes one second:
+
+```js
+// test.js
+before(
+    lolex.install();
+);
+// ...
+
+it('should call after one second', () => {
+    const spy = sinon.spy();
+    maker.callAfterOneSecond(spy);
+
+    // callback is not called immediately
+    assert.ok(!spy.called);
+
+    // but it is called synchronously after the clock is fast forwarded
+    clock.tick(1000);
+    assert.ok(spy.called); // PASS
+});
+```
+
+The same approach can be used to test an `async` function:
+
+```js
+module.exports.asyncReturnAfterOneSecond = async () => {
+    // util.promisify is not used deliberately
```

Comment:
> When it is used though, the test fails.

 Make it a proxy to `util.promisify` or call clock.install() before `util.promisify` is called (because then it holds a reference to the original setTimeout):

```js
module.exports. asyncReturnAfterOneSecond = async () => {
  const setTimeoutPromise = util.promisify(setTimeout); //
  // ...
}

```

Reply:
Nope, even with `lolex.install` call before `util.promisify`, tests of this function fail.

What did you mean by proxying somehting to `util.promisify`? I'll check it as well.

Reply:
I think this needs to be fixed in lolex. I'll make an issue.

Reply:
For reference, the issue is sinonjs/lolex#227